cmake/load_morpheus_utils.cmake

```diff
@@ -0,0 +1,49 @@
+# =============================================================================
+# SPDX-FileCopyrightText: Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+
+include_guard()
+include(FetchContent)
+
+# Fetch morpheus utilities -- don't use CPM, RAPIDS_CPM, or other external libraries here so
+#   we are only relying on CMake to get our core utilities.
+function(find_and_configure_morpheus_utils version)
+  list(APPEND CMAKE_MESSAGE_CONTEXT "morpheus_utils")
+
+  set(fetchcontent_tmp "${FETCHCONTENT_BASE_DIR}")
+  set(FETCHCONTENT_BASE_DIR "${CMAKE_SOURCE_DIR}/.cache") # default location
+  if (${CPM_SOURCE_CACHE})
+    set(FETCHCONTENT_BASE_DIR "${CPM_SOURCE_CACHE}")
+  endif()
+
+  FetchContent_Declare(
```

Comment:
Here is where we need to have a discussion on FetchContent vs Git Submodules.

I don't regularly touch files in the cmake directory, but with a submodule that is initialized and updated, it feels like a folder which I can navigate around and possibly edit.

With FetchContent, a copy is fetched and put in the build path. I think you can still edit that local copy and it maintains its git origins so you can push PRs back to the source should any changes be made.

My preference here would be a submodule, but I think it's something we need to evaluate the pros and cons and make a group decision.

Reply:
I don't think I have any strong opinion here, although I'm more or less on board with using sub-modules since it reduces dependencies.